lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -650,44 +651,28 @@ def compute_initial_guess(
             initial_guess = [0, 1, 0] + np.repeat(
                 np.sqrt(1. / no_of_realizations), no_of_realizations).tolist()
         elif estimate_coefficients_from_linear_model_flag:
+            truth_flattened = flatten_ignoring_masked_data(truth.data)
             if predictor_of_mean_flag.lower() == "mean":
-                # Find all values that are not NaN.
```

Comment:
Yes we have. I don't think we need to support this case any more, and it will require some more changes to support it properly. In both the calculate initial guess and in the CRPS calculation you need flattened arrays that are the same length for both the truth and historic forecasts. This means that if you have a missing data point in the truth data, then you need to remove the same point in the historic forecasts, and vice versa.